src/test/java/uk/gov/pay/adminusers/unit/service/ServiceResourceUpdateTest.java

```diff
@@ -0,0 +1,312 @@
+        when(mockedServiceDao.merge(thisServiceEntity)).thenReturn(thisServiceEntity);
+
+        Response response = resources.target(format(API_PATH, thisServiceEntity.getExternalId()))
+                .request()
+                .method("PATCH", Entity.json(jsonPayload));
+
+        assertThat(response.getStatus(), is(400));
+
+        String body = response.readEntity(String.class);
+        JsonPath json = JsonPath.from(body);
+
+        assertThat(json.getList("errors"), hasSize(1));
+        assertThat(json.getList("errors"), containsInAnyOrder("Field [path] is required"));
+    }
+
+    @Test
+    public void shouldError400_ifFieldOpMissing() {
+
+        ServiceEntity thisServiceEntity = ServiceEntityBuilder.aServiceEntity().build();
+        String externalId = thisServiceEntity.getExternalId();
+
+        String jsonPayload = fixture("fixtures/resource/service/patch/update-name-missing-op.json");
+        when(mockedServiceDao.findByExternalId(externalId)).thenReturn(Optional.of(thisServiceEntity));
+        when(mockedServiceDao.merge(thisServiceEntity)).thenReturn(thisServiceEntity);
+
+        Response response = resources.target(format(API_PATH, thisServiceEntity.getExternalId()))
+                .request()
+                .method("PATCH", Entity.json(jsonPayload));
+
+        assertThat(response.getStatus(), is(400));
+
+        String body = response.readEntity(String.class);
+        JsonPath json = JsonPath.from(body);
+
+        assertThat(json.getList("errors"), hasSize(1));
+        assertThat(json.getList("errors"), containsInAnyOrder("Field [op] is required"));
+    }
+
+    @Test
+    public void shouldSuccess_whenAddGatewayAccountIds_whereNoGatewayAccountIds() {
+        ServiceEntity thisServiceEntity = ServiceEntityBuilder.aServiceEntity().build();
+        String externalId = thisServiceEntity.getExternalId();
+
+        assertThat(thisServiceEntity.getGatewayAccountIds(), is(empty()));
+
+        String jsonPayload = fixture("fixtures/resource/service/patch/update-gateway-account-id.json");
+        when(mockedServiceDao.findByExternalId(externalId)).thenReturn(Optional.of(thisServiceEntity));
+        when(mockedServiceDao.merge(thisServiceEntity)).thenReturn(thisServiceEntity);
+        when(mockedServiceDao.checkIfGatewayAccountsUsed(Collections.singletonList("1014748185"))).thenReturn(false);
+
+        Response response = resources.target(format(API_PATH, thisServiceEntity.getExternalId()))
+                .request()
+                .method("PATCH", Entity.json(jsonPayload));
+
+        assertThat(response.getStatus(), is(200));
+
+        String body = response.readEntity(String.class);
+        JsonPath json = JsonPath.from(body);
+
+        assertThat(json.getList("gateway_account_ids"), contains("1014748185"));
+    }
+
+    @Test
+    public void shouldSuccess_whenAddGatewayAccountIds_whereThereIsGatewayAccountIds() {
```

Comment:
This is the equivalent of `ServiceResourceUpdateTest.shouldSuccess_whenAddGatewayAccountIds()`

Reply:
👍